src/components/ServiceCatalog/ServiceCatalog.tsx

```diff
@@ -387,7 +387,7 @@ const ServiceCatalog = ({ isDisabled }: Props) => {
     } catch (e) {
       api.setError(errorMessage(e));
     }
-  }, []);
+  }, [auth, signupData]);
```

Comment:
@mfrancisc I would encourage you to follow up with some tests to confirm we do not regress this logic in the future.